homeassistant/components/modbus/base_platform.py

```diff
@@ -188,10 +187,10 @@ def _swap_registers(self, registers: list[int], slave_count: int) -> list[int]:
             registers.reverse()
         return registers
 
-    def __process_raw_value(self, entry: float | int | str) -> float | int | str:
+    def __process_raw_value(self, entry: float | int | str) -> float | int | str | None:
         """Process value from sensor with NaN handling, scaling, offset, min/max etc."""
         if self._nan_value and entry in (self._nan_value, -self._nan_value):
-            return STATE_UNAVAILABLE
+            return None
```

Comment:
Do we want to use unknown state instead of unavailable state?

Reply:
I thought about that, but the state is known, since NAN is returned, which basically means the physical sensor have a problem, so my preference is STATE_UNAVAILABLE

Reply:
How is unavailable state set now? If I'm reading the code right, returning None here will be expressed as unknown state.

Reply:
Is it thus line?

https://github.com/home-assistant/core/blob/d955319cbad93aef92015bedd4cab36e4044a3bf/homeassistant/components/modbus/sensor.py#L137

Reply:
Yes.

When pymodbus (the library) encounters an illegal value it returns None, which have always lead the the sensor being set unavailable.

with this PR "NAN" (which is a legal value) is converted to None and thus handled identically.

Reply:
Thanks! 👍